components/Nav/Navbar/index.tsx

```diff
@@ -1,36 +1,204 @@
-import React from 'react';
+import React, { useCallback, useContext, useEffect, useState } from 'react';
 import Link from 'next/link';
 import { useRouter } from 'next/router';
 
 import shallow from 'zustand/shallow';
 import { Container } from '~/components/General/Container';
+import Hide from '~/components/General/Hide';
 import Show from '~/components/General/Show';
+import { LauncherToggle } from '~/components/Nav/Navbar/Buttons';
+import HamburgerMenu from '~/components/Nav/Navbar/MobileMenus/HamburgerMenu';
+import LauncherMenu from '~/components/Nav/Navbar/MobileMenus/LauncherMenu';
+import { PopoutButtons } from '~/components/Nav/Navbar/Popouts/Buttons';
+import TracerNavLogo from '~/components/Nav/Navbar/TracerNavLogo';
+import { NavContext, NavContextProvider } from '~/context/NavContext';
 import { useStore } from '~/store/main';
 import { selectWeb3Info } from '~/store/Web3Slice';
 
-import HeaderSiteSwitcher from './HeaderSiteSwitcher';
+import { classNames } from '~/utils/helpers';
 import AccountDropdown from './AccountDropdown';
-import MobileMenu from './MobileMenu';
+import MobileMenu from './MobileMenus/MobileNav';
 import NetworkDropdown from './NetworkDropdown';
-import { classNames } from '~/utils/helpers';
-import ThemeSwitcher from './ThemeSwitcher';
-import VersionToggle from './VersionToggle';
-
-import RevisitOnboard from '/public/img/general/onboard-revisit.svg';
+import HelpIconSVG from '/public/img/general/onboard-revisit.svg';
 
 const NavBar: React.FC<{
     setShowOnboardModal?: React.Dispatch<React.SetStateAction<boolean>>;
 }> = ({ setShowOnboardModal }) => {
+    return (
+        <NavContextProvider>
+            <NavBarContent setShowOnboardModal={setShowOnboardModal} />
+        </NavContextProvider>
+    );
+};
+
+const NavBarContent: React.FC<{
+    setShowOnboardModal?: React.Dispatch<React.SetStateAction<boolean>>;
+}> = ({ setShowOnboardModal }) => {
+    const { navMenuOpen, setNavMenuOpen, launcherMenuOpen, setLauncherMenuOpen } = useContext(NavContext);
+    const [navBackdrop, setNavBackdrop] = useState<boolean>(true);
+    const routes = useRouter().asPath.split('/');
+    const route = routes[1];
```

Comment:
is the intention of this to get the last part of the url? if so we can do `route[routes.length - 1]` instead?

looks like this could potentially not work as intended if the urls are longer/shorter in the future

Reply:
this was leftover from the previous navbar - though this solution works better. added